pkg/blockstorage/azure/auth.go

```diff
@@ -0,0 +1,94 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/kanisterio/kanister/pkg/blockstorage"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// currently avaialble types: https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authorization
+// to be available with azidentity: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-credential-types
+// to faciliate future migration, only the common ones between two SDKs are listed
+const (
+	CredTypeManagedIdentity   string = "cred-type-msi"
+	CredTypeClientSecret      string = "cred-type-client-secret"
+	CredTypeClientCertificate string = "cred-type-client-certificate"
+)
+
+var availableCredsType = sets.NewString(CredTypeManagedIdentity)
```

Comment:
Is this setting MSI as default?

Reply:
Not sure why a set is being used. Where are we building the set with available types?

Reply:
It does not set MSI as default, just a list of 1 since `Authenticated()` was only added for MSI. However I think it's confusing to only have MSI and not ClientCreds, so ClientCreds is added:
```go
var availableCredsType = sets.NewString(CredTypeManagedIdentity, CredTypeClientSecret)
```

a `set` is used so I can use an interface such as `Has()` to quickly search for supported credential types. When more credentials type is added in the future, only the `set` initialization needs to be extended